tracer/src/Datadog.Tracer.Native/debugger_method_rewriter.cpp

```diff
@@ -837,7 +1250,25 @@ HRESULT DebuggerMethodRewriter::Rewrite(RejitHandlerModule* moduleHandler,
     mdToken exceptionToken = mdTokenNil;
     mdToken callTargetReturnToken = mdTokenNil;
     ILInstr* firstInstruction;
-    hr = debuggerTokens->ModifyLocalSigAndInitialize(&rewriterWrapper, caller, &callTargetStateIndex, &exceptionIndex,
+
+    auto isAsyncMethodProbeHr = IsAsyncMethodProbe(module_metadata.metadata_import, caller);
+    IfFailRet(isAsyncMethodProbeHr);
+    TypeSignature methodReturnType{};
+    if (isAsyncMethodProbeHr == S_OK)
```

Comment:
`IsAsyncMethodProbe` should return a `bool`, using `S_OK` to denote `true` seems a tad confusing (and completely unnecessary) in this context. The fact that a lot of `COM` objects use this horrendous pattern doesn't mean we have to use it ourselves ;)

Reply:
The method have to use COM, it happens in the call to `IsTypeImplementIAsyncStateMachine`, because we might have a FAIL result we need HRESULT in addition to the bool result. We can return HRESULT and pass `out bool` argument or do the opposite or leave it like this. WDYT?

Reply:
Fixed in 5a103fbe57cedbc9e13568d4372c82a921c46e26 by using out bool arg